addons/website_profile/static/src/js/website_profile.js

```diff
@@ -25,12 +25,23 @@ publicWidget.registry.websiteProfile = publicWidget.Widget.extend({
     _onSendValidationEmailClick: function (ev) {
         ev.preventDefault();
         var $element = $(ev.currentTarget);
+
+        if (this.emailRequested) {
+            return;
+        }
+
+        $element[0].style.cursor = "progress";
+        this.emailRequested = true;
         this.rpc('/profile/send_validation_email', {
             'redirect_url': $element.data('redirect_url'),
         }).then(function (data) {
             if (data) {
                 window.location = $element.data('redirect_url');
             }
+        }).catch((e) => {
+            this.emailRequested = null;
```

Comment:
Nitpicking:

```suggestion
            this.emailRequested = false;
```

Reply:
I would also add a comment to make it clear that we want to only allow requesting an email once per page load. Maybe here or in the method's jsdoc.